src/trie/proof_node_decode.rs

```diff
@@ -19,19 +19,36 @@ use super::nibble;
 use core::{fmt, iter, slice};
 
 /// Decodes a node value found in a proof into its components.
+///
+/// This can decode nodes no matter their version.
 pub fn decode(mut node_value: &[u8]) -> Result<Decoded, Error> {
     if node_value.is_empty() {
         return Err(Error::Empty);
     }
 
-    let has_children = (node_value[0] & 0x80) != 0;
-    let has_storage_value = (node_value[0] & 0x40) != 0;
+    let (has_children, storage_value_hashed, pk_len_first_byte_bits) = match node_value[0] >> 6 {
+        0b00 => {
+            if (node_value[0] >> 5) == 0b001 {
```

Comment:
why not `match`?

Reply:
You'd need to write something like:

```
match node_value[0] {
    v if (v >> 5) == 0b001 => ...
    v if (v >> 4) == 0b0001 => ...
    0 => ...
    _ => 
}
```

And I just find the ifs more readable